components/Shipping/basic_usage.rst

```diff
@@ -0,0 +1,657 @@
+        $volumeRateCalculator = new VolumeRateCalculator();
+
+        shippable->setShippingVolume(100);
+        $volumeRateCalculator->calculate($shipment, $configuration); // returns 4000
+
+        $shippable->setShippingVolume(20);
+        $volumeRateCalculator->calculate($shipment, $configuration); // returns  800
+
+WeightRateCalculator
+~~~~~~~~~~~~~~~~~~~~
+
+**WeightRateCalculator** charges amount rate per weight.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Shipment;
+        use Sylius\Component\Shipping\Model\ShipmentItem;
+        use Sylius\Component\Shipping\Calculator\WeightRateCalculator;
+
+        $configuration = array('fixed' => 200, 'variable' => 500, 'division' => 5);
+        // this configuration should be defined in shipping method allowed for shipment
+        $weightRateCalculator = new WeightRateCalculator();
+
+        $shippable = new ShippableObject();
+
+        $shipmentItem = new ShipmentItem();
+        $shipmentItem->setShippable($shippable);
+
+        $shipment = new Shipment();
+        $shipment->addItem($shipmentItem);
+
+        $shippable->setShippingWeight(100);
+        $weightRateCalculator->calculate($shipment, $configuration); // returns 10200
+
+        $shippable->setShippingWeight(10);
+        $weightRateCalculator->calculate($shipment, $configuration); // returns 1200
+
+Resolvers
+---------
+
+.. _method-resolver:
+
+MethodsResolver
+~~~~~~~~~~~~~~~
+
+**MethodResolver** provides information about the number of methods allowed for shipment object.
+First you need to create a few instances of **ShippingCategory** class:
+
+.. code-block:: php
+
+    <?php
+
+    use Sylius\Component\Shipping\Model\ShippingCategory;
+
+    $shippingCategory = new ShippingCategory();
+    $shippingCategory->setName('Regular');
+    $shippingCategory1 = new ShippingCategory();
+    $shippingCategory1->setName('Light');
+    $shippingCategory2 = new ShippingCategory();
+    $shippingCategory2->setName('Heavy');
+
+
+Next you have to create a repository w which holds a few instaces of **ShippingMethod**. An InMemoryRepository,
+which holds a collection of **ShippingMethod** objects, was used. The configuration is shown below:
+
+.. code-block:: php
+
+    <?php
+
+            // ...
+            // notice:
+            // $categories = array($shippingCategory, $shippingCategory1, $shippingCategory2);
+
+            $firstTranslate = new ShippingMethodTranslation();
+            $firstTranslate->setLocale('en');
+            $firstTranslate->setName('First method');
+
+            $secondTranslate = new ShippingMethodTranslation();
+            $secondTranslate->setLocale('en');
+            $secondTranslate->setName('Second method');
+
+            $thirdTranslate = new ShippingMethodTranslation();
+            $thirdTranslate->setLocale('en');
+            $thirdTranslate->setName('Third method');
```

Comment:
These are not good variable names. :P also, you don't really need to bother with translations. They will use ``en_US`` as default, so you can just do new ShippingMethod and that's it. We should keep these examples simple. This is not really simple. :>